### PR TITLE
gigasecond: add exercise

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

**Comment by colinleach** [comment]
I don't know if it matters, but there is a mismatch between `tests.toml` and the tests. The "does not mutate the input" test is listed but not implemented.

FWIW, this is pretty common across tracks.

**Review by colinleach — APPROVED** [review]
Nobody could complain that the example implementation is too verbose!

File: exercises/practice/gigasecond/.meta/tests.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+description = "third test for date only specification of time"
+
+[c9d89a7d-06f8-4e28-a305-64f1b2abc693]
+description = "full time specified"
+
+[09d4e30e-728a-4b52-9005-be44a58d9eba]
+description = "full time with day roll-over"
+
+[fcec307c-7529-49ab-b0fe-20309197618a]
+description = "does not mutate the input"
```
Comment (ErikSchierboom) [inline exercises/practice/gigasecond/.meta/tests.toml:28]:
```suggestion
description = "does not mutate the input"
include = false
```

**Comment by ErikSchierboom** [comment]
@colinleach Updated!

**Pull request merged**